src/main/java/com/coveros/selenified/element/check/Contains.java

```diff
@@ -0,0 +1,281 @@
+        }
+        return actualClass;
+    }
+
+    /**
+     * Verifies that the element contains the provided expected attribute. If
+     * the element isn't present, this will constitute a failure, same as a
+     * mismatch. This information will be logged and recorded, with a screenshot
+     * for traceability and added debugging support.
+     *
+     * @param expectedAttribute - the attribute to check for
+     */
+    void attribute(String expectedAttribute);
+
+    default String[] checkAttribute(String expectedAttribute, double waitFor, double timeTook) {
+        // record the action and get the attributes
+        String[] allAttributes = getAttributes(expectedAttribute, "with", waitFor);
+        // record the result
+        if (allAttributes == null || !Arrays.asList(allAttributes).contains(expectedAttribute)) {
+            getOutputFile().recordActual(
+                    getElement().prettyOutputStart() + " does not contain the attribute of <b>" + expectedAttribute + "</b>" +
+                            ONLYVALUE + Arrays.toString(allAttributes) + "</b>", timeTook, Success.FAIL);
+        } else {
+            getOutputFile().recordActual(getElement().prettyOutputStart() + " contains the attribute of <b>" + expectedAttribute + "</b>",
+                    timeTook, Success.PASS);
+        }
+        return allAttributes;
+    }
+
+    /**
+     * Verifies that the element's text contains the provided expected text. If
+     * the element isn't present, this will constitute a failure, same as a
+     * mismatch. This information will be logged and recorded, with a screenshot
+     * for traceability and added debugging support.
+     *
+     * @param expectedText the expected value of the element
+     */
+    void text(String expectedText);
+
+    default String checkText(String expectedText, double waitFor, double timeTook) {
+        // record the action
+        getOutputFile().recordAction(getElement().prettyOutput() + HASTEXT + expectedText + "</b>", waitFor);
+        // get the value
+        String elementValue = getElement().get().text();
+        // record the result
+        if (elementValue == null || !elementValue.contains(expectedText)) {
+            getOutputFile().recordActual(getElement().prettyOutputStart() + TEXT + elementValue + "</b>", timeTook, Success.FAIL);
+        } else {
+            getOutputFile().recordActual(getElement().prettyOutputStart() + TEXT + elementValue + "</b>", timeTook, Success.PASS);
+        }
+        return elementValue;
+    }
+
+    /**
+     * Verifies that the element's value contains the provided expected value.
+     * If the element isn't present or an input, this will constitute a failure,
+     * same as a mismatch. This information will be logged and recorded, with a
+     * screenshot for traceability and added debugging support.
+     *
+     * @param expectedValue the expected value of the element
+     */
+    void value(String expectedValue);
```

Comment:
![BLOCKER](https://sonarsource.github.io/sonar-github/severity-blocker.png 'Severity: BLOCKER') Rename method "value" to prevent any misunderstanding/clash with field "VALUE" defined in interface "com.coveros.selenified.element.check.Check". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://localhost:9000/sonar/coding_rules#rule_key=squid%3AS1845)